e2e-test/src/main.rs

```diff
@@ -240,7 +246,7 @@ async fn main() {
 /// - *device*: the Astarte SDK instance to use for the test.
 /// - *test_cfg*: struct containing configuration settings for the tests.
 async fn test_datastream_device_to_server(
-    device: &AstarteDeviceSdkMemory,
+    device: &impl Device,
```

Comment:
Is the reference (`&`) necessary?

Reply:
Yes, we can't take ownership of the device here and we shouldn't since there aren't consuming methods.